File: langgraph/result/qwen/14b-py2js/46.js
function fib4(n) {
    const results = [0, 0, 2, 0];
    if (n < 4) {
        return results[n];
    }

    for (let i = 4; i <= n; i++) {
        results.push(results[results.length - 1] + results[results.length - 2] + results[results.length - 3] + results[results.length - 4]);
        results.shift();
    }

    return results[results.length - 1];
}

File: langgraph/data/semantic_equiv/46.py
def calculate_custom_fibonacci(m: int):
    sequence = [0, 0, 2, 0]
    if m < 4:
        return sequence[m]

    index = 4
    while index <= m:
        next_value = sequence[-1] + sequence[-2] + sequence[-3] + sequence[-4]
        sequence.append(next_value)
        sequence.pop(0)
        index += 1

    return sequence[-1]

# method name: fib4 -> calculate_custom_fibonacci
# parameter name: n -> m
# variable name: results -> sequence
# changed for loop to while loop with index
# added index variable to control the loop